turbopack/crates/turbopack/src/module_options/module_options_context.rs

```diff
@@ -112,36 +112,43 @@ pub struct JsxTransformOptions {
 #[derive(Clone, Default)]
 #[serde(default)]
 pub struct ModuleOptionsContext {
-    pub enable_typeof_window_inlining: Option<TypeofWindow>,
-    pub enable_jsx: Option<Vc<JsxTransformOptions>>,
+    pub ecmascript: EcmascriptOptionsContext,
+    pub css: CssOptionsContext,
+
     pub enable_postcss_transform: Option<Vc<PostCssTransformOptions>>,
     pub enable_webpack_loaders: Option<Vc<WebpackLoadersOptions>>,
-    /// Follow type references and resolve declaration files in additional to
-    /// normal resolution.
-    pub enable_types: bool,
-    pub enable_typescript_transform: Option<Vc<TypescriptTransformOptions>>,
-    pub decorators: Option<Vc<DecoratorsOptions>>,
-    pub enable_mdx: bool,
-    /// This skips `GlobalCss` and `ModuleCss` module assets from being
-    /// generated in the module graph, generating only `Css` module assets.
-    ///
-    /// This is useful for node-file-trace, which tries to emit all assets in
-    /// the module graph, but neither asset types can be emitted directly.
-    pub enable_raw_css: bool,
     // [Note]: currently mdx, and mdx_rs have different configuration entrypoint from next.config.js,
     // however we might want to unify them in the future.
+    pub enable_mdx: bool,
     pub enable_mdx_rs: Option<Vc<MdxTransformOptions>>,
+
     pub preset_env_versions: Option<Vc<Environment>>,
-    /// Custom rules to be applied after all default rules.
-    pub custom_rules: Vec<ModuleRule>,
     pub execution_context: Option<Vc<ExecutionContext>>,
+    pub side_effect_free_packages: Vec<RcStr>,
+    pub tree_shaking_mode: Option<TreeShakingMode>,
+
+    pub special_exports: Option<Vc<Vec<RcStr>>>,
```

Comment:
This field was added since we last talked about it. Should it go into `context.ecmascript`?